spring-2013/hw1/src/CreditCardValidator.hs

```diff
@@ -0,0 +1,15 @@
+module CreditCardValidator (toDigits, toDigitsRev, doubleEveryOther) where
+
+toDigits :: Integer -> [Integer]
+toDigits xs
+    | xs <= 0 = []
+toDigits xs = map readChar . show $ xs
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev = reverse . toDigits
+
+readChar :: Read c => Char -> c
+readChar = read . (:[])
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther = reverse . zipWith (*) (cycle [1, 2]) . reverse
```

Comment:
Here's a cool version that I think Mike came up with: 

```haskell
reverse . zipWith ($) (cycle [id, (*2)]) . reverse 
```

It's zipping with `($)`.